client/layout/index.jsx

```diff
@@ -147,6 +147,23 @@ class Layout extends Component {
 	}
 
 	render() {
+		// This is temporary helper function until we have rolled out to 100% of customers.
+		const isNavUnificationEnabled = () => {
+			// Having the feature enabled by default in all environments, will let anyone use ?flags=-nav-unification to temporary disable it.
+			if ( 'production' !== process.env.NODE_ENV && ! config.isEnabled( 'nav-unification' ) ) {
```

Comment:
Why is the `'production' !== process.env.NODE_ENV` check needed? Would not that prevent a12s from disabling the nav unification when `NODE_ENV` is set to `production`?

Reply:
Yes indeed. Because of the fact that we now set `"nav-unification": false` in production we need an escape mechanism so that it is still enabled for all a12s who don't use `?flags=-nav-unification`. Again, the `isAutomatticTeamMember` is the only consistent way of having the feature enabled across all infrastructures ( Calypso, Simple, Atomic) so enabling the feature depending on this check on all Calypso environments should be the best call.

Reply:
Do we know if `NODE_ENV` is set to `production` for a12s visiting wordpress.com while proxied? That's technically the staging environment, so I wonder if in these cases `NODE_ENV` has a different value like `staging` or `stage`.

I have the feeling that the current feature flags system is not suited for our needs. Why don't we completely avoid it and use a customized one which is based on [`sessionStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage)? 

```js
const isNavUnificationEnabled = () => {
  const STORAGE_KEY = 'nav_unification_enabled';

  // Explicitly disabled by query param.
  if ( ( new URL( document.location ) ).searchParams.get( 'disable-nav-unification' ) ) {
    window.sessionStorage.setItem( STORAGE_KEY, false );
    return false;
  }
  
  // Check if has been already enabled/disabled on current session.
  const isEnabled = window.sessionStorage.getItem( STORAGE_KEY );
  if ( isEnabled !== null ) {
    return isEnabled;
  }

  // Enable for all a12s.
  if ( this.props.teams.length && isAutomatticTeamMember( this.props.teams ) ) {
    window.sessionStorage.setItem( STORAGE_KEY, true );
    return true;
  }
  
  return false;
};
```

Reply:
`NODE_ENV` is set to `staging` for a12s visiting wordpress.com while proxied and that's why the will get the chance to disable the feature 

```
if ( 'production' !== process.env.NODE_ENV && ! config.isEnabled( 'nav-unification' ) ) {
  return false;
}
```

Why doesn't the flag system help us here? Instead of using it per environment, we are using it internally (with a custom condition) so that we avoid bugs that a "new system" may include. Also, if we decide changing it we would also need to change all `nav-unification` occurrences which depend on it. I am not that such a temporary solution deserves that effort. Coding it seems super simple but testing it is certainly not.

Reply:
> NODE_ENV is set to staging for a12s visiting wordpress.com

Cool, thanks for confirming. I don't know why I was under the impression that we were using `production` as well on staging. The `'production' !== process.env.NODE_ENV && ! config.isEnabled( 'nav-unification' )` check makes sense now 👍